src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,53 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+
+    let shouldInclude = true;
+
+    for (const animal of Animal.alive) {
+      if (animal.name === this.name) {
+        shouldInclude = false;
+      }
+    }
+
+    if (shouldInclude === true) {
+      Animal.alive.push(this);
+    }
+    Animal.alive.sort((a, b) => b.health - a.health);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  bite(obj) {
+    if (obj instanceof Herbivore && obj.hidden === false) {
+      obj.health -= 50;
+    }
+
+    if (obj instanceof Herbivore && obj.health <= 0) {
+      Animal.alive.pop();
+    }
+  }
```

Comment:
I believe it looks better this way:
- what is `obj`? let's call it `victim` instead
- double-checking for `instanceof` – not OK, we can exit the function right at the beginning if needed
- you can avoid using `===` for types which is convertible to boolean
```
bite(victim) {
  if (!(victim instanceof Herbivore)) {
    return;
  }
  
  if (!victim.hidden) {
    victim.health -= 50;
  }

  if (!victim.health) {
    Animal.alive.filter(...);
  }
}
```